Swift/Sources/StateMachine/StateMachine.swift

```diff
@@ -1,52 +1,50 @@
 //
-//  Created by Christopher Fuller on 12/21/19.
-//  Copyright © 2019 Tinder. All rights reserved.
+//  Copyright (c) 2018, Match Group, LLC
+//  BSD License, see LICENSE file for details
 //
 
 open class StateMachine<State: StateMachineHashable, Event: StateMachineHashable, SideEffect> {
 
-    public typealias FunctionBuilder = StateMachineTypes.StateMachineFunctionBuilder
-
-    public typealias Definition = StateMachineTypes.StateMachineDefinition<State, Event, SideEffect>
-    public typealias InitialState = StateMachineTypes.InitialState<State>
-    public typealias Component = StateMachineTypes.Component<State, Event, SideEffect>
-
-    private typealias States = [State.HashableIdentifier: Events]
-    private typealias Events = [Event.HashableIdentifier: Action.Factory]
-
-    private typealias EventHandler = StateMachineTypes.EventHandler<State, Event, SideEffect>
-    private typealias Action = StateMachineTypes.Action<State, Event, SideEffect>
-
     public enum Transition {
 
         public typealias Result = Swift.Result<Valid, Error>
         public typealias Callback = (_ result: Result) -> Void
 
         public struct Valid: CustomDebugStringConvertible {
 
+            public var debugDescription: String {
+                guard let sideEffect: SideEffect = sideEffect
+                else { return "fromState: \(fromState), event: \(event), toState: \(toState), sideEffect: nil" }
+                return "fromState: \(fromState), event: \(event), toState: \(toState), sideEffect: \(sideEffect)"
+            }
```

Comment:
could do a nil coalescing operator to default to "nil" string if you wanna reduce lines

Reply:
_updated_

Reply:
Oh, wait. Sorry. That errors with …
```
binary operator '??' cannot be applied to operands of type 'SideEffect?' and 'String'
```
Going to revert that now.